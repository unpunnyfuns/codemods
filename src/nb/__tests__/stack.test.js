import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/stack', () => {
  it('migrates HStack to Stack with full prop mapping', () => {
    const output = testTransform('nb/stack', 'nb/stack', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Stack',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
