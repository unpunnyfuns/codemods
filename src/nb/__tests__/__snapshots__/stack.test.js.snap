// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/stack > migrates HStack to Stack with full prop mapping 1`] = `
"import { SomeOtherComponent } from '@org/common/src/other'
import { Stack } from '@hb-frontend/app/src/components/nordlys/Stack';
import { radius } from '@hb-frontend/nordlys';
import { StyleSheet } from 'react-native';

export function MyComponent() {
  return (
    <>

      {/* gap={2} */
      }
      <Stack direction='horizontal' style={styles.hstack0}>
        <SomeOtherComponent />
      </Stack>
      <Stack testID="test-stack" direction='horizontal' style={styles.hstack1}>
        <SomeOtherComponent />
      </Stack>

      {/* reversed={true}
        divider={...} */
      }
      <Stack onPress={() => {}} direction='horizontal' style={styles.hstack2}>
        <SomeOtherComponent />
      </Stack>
    </>
  );
}

const styles = StyleSheet.create({
  hstack0: {
    margin: 4,
    alignItems: 'flex-start',
    borderRadius: radius.md
  },

  hstack1: {
    marginTop: 8,
    paddingHorizontal: 3,
    paddingVertical: 2,
    justifyContent: 'space-between'
  },

  hstack2: {
    marginHorizontal: 5,
    marginBottom: 1
  }
});
"
`;
